Xamarin.Forms.Platform.MacOS/Renderers/EntryRenderer.cs

```diff
@@ -9,15 +10,50 @@ public class EntryRenderer : ViewRenderer<Entry, NSTextField>
 		class FormsNSTextField : NSTextField
 		{
 			public EventHandler<BoolEventArgs> FocusChanged;
+
+			bool _windowEventsSet;
+
 			public override bool ResignFirstResponder()
 			{
-				FocusChanged?.Invoke(this, new BoolEventArgs(false));
 				return base.ResignFirstResponder();
 			}
+
 			public override bool BecomeFirstResponder()
 			{
 				FocusChanged?.Invoke(this, new BoolEventArgs(true));
-				return base.BecomeFirstResponder();
+
+				var result = base.BecomeFirstResponder();
+
+				if (!_windowEventsSet)
+				{
+					_windowEventsSet = true;
+					Window.DidResignKey += HandleWindowDidResignKey;
+					Window.DidBecomeKey += HandleWindowDidBecomeKey;
+				}
+
+				return result;
+			}
+
+			public override void DidEndEditing(NSNotification notification)
+			{
+				if (CurrentEditor != Window.FirstResponder)
+				{
+					FocusChanged?.Invoke(this, new BoolEventArgs(false));
```

Comment:
i wonder if we could get in a place that we fire the event multiple times here, and also fire the `FocusChanged` on `HandleWindowDidResignKey`